project/src/components/favorite-list/favotite-list.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getStarRaiting } from '../../common';
+
+function FavoriteList(props) {
+  const { offers } = props;
+  const citys = [];
+
+  for (let i = 0; i < offers.length; i++) {
+    if (!citys.includes(offers[i].city.name) && offers[i].isFavorite) {
+      citys.push(offers[i].city.name);
+    }
+  }
+
+  if (citys.length === 0) {
+    return (<section className='favorites favorites--empty'>
+      <h1 className='visually-hidden'>Favorites (empty)</h1>
+      <div className='favorites__status-wrapper'>
+        <b className='favorites__status'>Nothing yet saved.</b>
+        <p className='favorites__status-description'>Save properties to narrow down search or plan your future trips.</p>
+      </div></section>);
+  }
+
+  return (
+    <section className='favorites'>
+      <h1 className='favorites__title'>Saved listing</h1>
+      <ul className='favorites__list'>
+
+        {citys.map((city, idx) => (
+          <li className='favorites__locations-items' key={idx.toString()}>
+            <div className='favorites__locations locations locations--current'>
+              <div className='locations__item'>
+                <a className='locations__item-link' href='#'>
+                  <span>{city}</span>
+                </a>
+              </div>
+            </div>
+            <div className='favorites__places'>
+
+              {offers.map((offer, idx1) => {
+                if (offer.city.name === city) {
+                  return (<article className='favorites__card place-card' key={idx1.toString()}>
```

Comment:
```suggestion
                  return (<article className='favorites__card place-card' key={offer.id}>
```
Нельзя использовать индекс массива для ключа. Раньше мы этот делали тк не было выхода. Сейчас есть